app/test/features/chat_ng/widgets/chat_messages_list_test.dart

```diff
@@ -0,0 +1,406 @@
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => animatedListKey),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      expect(animatedListKey.currentState?.widget.initialItemCount, equals(3));
+
+      // Verify initial message order
+      expect(notifier.state.messageList, equals(['msg0', 'msg1', 'msg2']));
+
+      // add new message
+      final updatedState = initialState.copyWith(
+        messageList: [...initialState.messageList, 'newMessage'],
+      );
+      notifier.state = updatedState;
+
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 300));
+
+      // verify the list was updated
+      expect(animatedListKey.currentState?.widget.initialItemCount, equals(4));
+
+      // Verify final message order
+      expect(
+        notifier.state.messageList,
+        equals(['msg0', 'msg1', 'msg2', 'newMessage']),
+      );
+    });
+
+    testWidgets('scroll to end animates to the bottom of the list', (
+      WidgetTester tester,
+    ) async {
+      final messagesState = ChatRoomState(
+        messageList: List.generate(20, (index) => 'msg$index'),
+        loading: const ChatRoomLoadingState.loaded(),
+      );
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(messagesState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      final scrollController =
+          tester.widget<AnimatedList>(find.byType(AnimatedList)).controller;
+      final chatMessagesWidget = find.byType(ChatMessages);
+
+      // scroll up to make the button visible
+      scrollController?.position.jumpTo(100.0);
+
+      await tester.pump();
+
+      final initialState =
+          tester.state(chatMessagesWidget) as ConsumerState<ChatMessages>;
+      expect((initialState as dynamic).showScrollToBottom, isTrue);
+
+      final finalState =
+          tester.state(chatMessagesWidget) as ConsumerState<ChatMessages>;
+
+      await (finalState as dynamic).scrollToEnd();
+
+      await tester.pump();
+      //wait for the animation to complete
+      await tester.pumpAndSettle();
+
+      expect(scrollController?.position.pixels, equals(0.0));
+      // button should be hidden
+      expect((finalState as dynamic).showScrollToBottom, isFalse);
+    });
+
+    testWidgets('preserves correct message order when displayed in reverse', (
```

Comment:
does this matter to us (at all)? this feels like a pointless test of whether the builtin `reverse` of dart lists work. Let's drop this test.